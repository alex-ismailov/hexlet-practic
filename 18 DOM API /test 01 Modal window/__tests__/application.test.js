// import 'js-polyfills/html';
// import 'js-polyfills/dom';
import fs from 'fs';
import path from 'path';
import prettier from 'prettier';

import run from '../src/application.js';

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => prettier.format(document.body.innerHTML, options);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', () => {
  const [button1, button2] = document.querySelectorAll('[data-toggle="modal"]');
  const [closeButton1, closeButton2] = document.querySelectorAll('[data-dismiss="modal"]');

  expect(getTree()).toMatchSnapshot();

  button1.click();
  expect(getTree()).toMatchSnapshot();

  closeButton1.click();
  expect(getTree()).toMatchSnapshot();

  button2.click();
  expect(getTree()).toMatchSnapshot();

  closeButton2.click();
  expect(getTree()).toMatchSnapshot();
});
