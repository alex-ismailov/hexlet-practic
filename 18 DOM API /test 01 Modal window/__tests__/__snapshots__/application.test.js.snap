// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal\\"
    >
        Launch demo modal 1
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 1
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>

    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal2\\"
    >
        Launch demo modal 2
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal2\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 2
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal\\"
    >
        Launch demo modal 1
    </button>
    <div
        class=\\"modal fade show\\"
        id=\\"exampleModal\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: block;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 1
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>

    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal2\\"
    >
        Launch demo modal 2
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal2\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 2
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal\\"
    >
        Launch demo modal 1
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: none;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 1
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>

    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal2\\"
    >
        Launch demo modal 2
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal2\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 2
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal\\"
    >
        Launch demo modal 1
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: none;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 1
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>

    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal2\\"
    >
        Launch demo modal 2
    </button>
    <div
        class=\\"modal fade show\\"
        id=\\"exampleModal2\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: block;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 2
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal\\"
    >
        Launch demo modal 1
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: none;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 1
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>

    <button
        type=\\"button\\"
        class=\\"btn btn-primary\\"
        data-toggle=\\"modal\\"
        data-target=\\"#exampleModal2\\"
    >
        Launch demo modal 2
    </button>
    <div
        class=\\"modal fade\\"
        id=\\"exampleModal2\\"
        tabindex=\\"-1\\"
        role=\\"dialog\\"
        aria-labelledby=\\"exampleModalLabel\\"
        aria-hidden=\\"true\\"
        style=\\"display: none;\\"
    >
        <div class=\\"modal-dialog\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\" id=\\"exampleModalLabel\\">
                        Modal title 2
                    </h5>
                    <button
                        type=\\"button\\"
                        class=\\"close\\"
                        data-dismiss=\\"modal\\"
                        aria-label=\\"Close\\"
                    >
                        <span aria-hidden=\\"true\\">×</span>
                    </button>
                </div>
                <div class=\\"modal-body\\">
                    ...
                </div>
                <!-- <div class=\\"modal-footer\\"> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button> -->
                <!--   <button type=\\"button\\" class=\\"btn btn-primary\\">Save changes</button> -->
                <!-- </div> -->
            </div>
        </div>
    </div>
</div>
"
`;
